Add MarkableTraits to FloatRect / FloatPoint / FloatSize

Constellation opened the pull request:
#### a1e8662310da9aff0f6ca3e3fb9a32202bb88c29
<pre>
Add MarkableTraits to FloatRect / FloatPoint / FloatSize
<a href="https://bugs.webkit.org/show_bug.cgi?id=262766">https://bugs.webkit.org/show_bug.cgi?id=262766</a>
rdar://116562787

Reviewed by Chris Dumez.

This patch adds MarkableTraits to FloatRect / FloatPoint / FloatSize.
So that we can use Markable&lt;FloatRect, FloatRect::MarkableTraits&gt; easily.
We also add `fabsConstExpr` to ensure that this is becoming `constexpr`.
And use `isNaNConstExpr` too. Unfortunately they are not constexpr until C++23.

* Source/WTF/wtf/MathExtras.h:
(WTF::fabsConstExpr):
* Source/WebCore/platform/graphics/FloatPoint.h:
(WebCore::FloatPoint::MarkableTraits::isEmptyValue):
(WebCore::FloatPoint::MarkableTraits::emptyValue):
(WebCore::FloatPoint::nanPoint):
(WebCore::FloatPoint::isNaN const):
* Source/WebCore/platform/graphics/FloatRect.h:
(WebCore::FloatRect::MarkableTraits::isEmptyValue):
(WebCore::FloatRect::MarkableTraits::emptyValue):
(WebCore::FloatRect::isNaN const):
* Source/WebCore/platform/graphics/FloatSize.h:
(WebCore::FloatSize::MarkableTraits::isEmptyValue):
(WebCore::FloatSize::MarkableTraits::emptyValue):
(WebCore::FloatSize::isZero const):
(WebCore::FloatSize::nanSize):
(WebCore::FloatSize::isNaN const):
* Tools/TestWebKitAPI/Tests/WebCore/FloatPointTests.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebCore/FloatRectTests.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebCore/FloatSizeTests.cpp:
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/268992@main">https://commits.webkit.org/268992@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b19acb46cd1371bb1f712cfec951a7d660b65702

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21225 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21560 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22278 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/23092 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19703 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/21464 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24837 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21765 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20934 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21448 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/21128 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18392 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23945 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/18287 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/19228 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/25584 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/18457 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19366 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19431 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23421 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/20607 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19948 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16958 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/24629 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19245 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/5831 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23515 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/25897 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2632 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19833 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5655 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
LGTM though it would have been nice to use the functionality you introduce in the same patch. Otherwise, it's just dead code at the moment.

Constellation committed:
Add MarkableTraits to FloatRect / FloatPoint / FloatSize
https://bugs.webkit.org/show_bug.cgi?id=262766
rdar://116562787

Reviewed by Chris Dumez.

This patch adds MarkableTraits to FloatRect / FloatPoint / FloatSize.
So that we can use Markable<FloatRect, FloatRect::MarkableTraits> easily.
We also add `fabsConstExpr` to ensure that this is becoming `constexpr`.
And use `isNaNConstExpr` too. Unfortunately they are not constexpr until C++23.

* Source/WTF/wtf/MathExtras.h:
(WTF::fabsConstExpr):
* Source/WebCore/platform/graphics/FloatPoint.h:
(WebCore::FloatPoint::MarkableTraits::isEmptyValue):
(WebCore::FloatPoint::MarkableTraits::emptyValue):
(WebCore::FloatPoint::nanPoint):
(WebCore::FloatPoint::isNaN const):
* Source/WebCore/platform/graphics/FloatRect.h:
(WebCore::FloatRect::MarkableTraits::isEmptyValue):
(WebCore::FloatRect::MarkableTraits::emptyValue):
(WebCore::FloatRect::isNaN const):
* Source/WebCore/platform/graphics/FloatSize.h:
(WebCore::FloatSize::MarkableTraits::isEmptyValue):
(WebCore::FloatSize::MarkableTraits::emptyValue):
(WebCore::FloatSize::isZero const):
(WebCore::FloatSize::nanSize):
(WebCore::FloatSize::isNaN const):
* Tools/TestWebKitAPI/Tests/WebCore/FloatPointTests.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebCore/FloatRectTests.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebCore/FloatSizeTests.cpp:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/268992@main
